firestore.rules

```diff
@@ -67,11 +74,18 @@ service cloud.firestore {
       allow write: if isGlobalAdmin(request.auth.token.email);
     }
 
-    match /recipients/{data=**} {
-      allow read: if isGlobalAdmin(request.auth.token.email) || isGlobalAnalyst(request.auth.token.email) || canAccessRecipient(request.auth.token);
+    match /recipients/{recipientId} {
+      allow read: if isGlobalAdmin(request.auth.token.email) || isGlobalAnalyst(request.auth.token.email) || canAccessRecipient(request.auth.token.email, resource.data);
       allow create: if isAdminUser(request.auth.token.email);
-      allow update: if isGlobalAdmin(request.auth.token.email) || canAccessRecipient(request.auth.token);
+      allow update: if isGlobalAdmin(request.auth.token.email) || canAccessRecipient(request.auth.token.email, resource.data);
       allow delete: if isGlobalAdmin(request.auth.token.email);
+
+      match /payments/{paymentId} {
+          allow read: if canAccessPayment(request.auth.token, recipientId);
+          allow create: if isAdminUser(request.auth.token.email);
+          allow update: if isGlobalAdmin(request.auth.token.email) || canAccessPayment(request.auth.token, recipientId);
```

Comment:
@ssandino, who should be allowed to update the payments? Only global admins? Organisations admins? Or everyone who as access to the payment (including the recipient)?

Reply:
On the app recipient should be able to change the status (e.g. set status that payment was not received). I'm not sure if it is safe to assume that app will allow only status change and we will be OK with that or should we create another document to link payment status with actual payment.